server/controllers/logger.ts

```diff
@@ -22,51 +22,55 @@ interface UserData {
   recommendation: string;
 }
 
-//func to insert and get data from the database
-export const insertUserData = async (
-  userData: UserData,
-): Promise<UserData | null> => {
-  const { data, error } = await supabase.from('user_data').insert([userData]);
+export const insertUserDataMiddleware = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  const userData: UserData = res.locals;
 
-  if (error) {
-    console.error('Error inserting data:', error);
-    return null;
+  if (!userData) {
+    return next(error);
```

Comment:
@nanyi-deng error is not defined here, you need to pass an object of type `ServerError` (check kenny's types file)